// dart-tools/dist/services/TaskService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskService = void 0;
const OpenAPI_1 = require("../core/OpenAPI");
const request_1 = require("../core/request");
class TaskService {
    /**
     * Create a new task
     * Record a new task that the user intends to do. This will save the task in Dart for later access, search, etc. By default the created task will be assigned to the user, with a status of to-do, with no parent, in the Active dartboard. More information can be included in the description.
     * @param requestBody
     * @returns WrappedTask
     * @throws ApiError
     */
    static createTask(requestBody) {
        return (0, request_1.request)(OpenAPI_1.OpenAPI, {
            method: "POST",
            url: "/tasks",
            body: requestBody,
            mediaType: "application/json",
        });
    }
    /**
     * Retrieve an existing task
     * Retrieve an existing task. This will return the task's information, including the title, dartboard, status, description and others.
     * @param id
     * @returns WrappedTask Success, including the retrieved task
     * @throws ApiError
     */
    static retrieveTask(id) {
        return (0, request_1.request)(OpenAPI_1.OpenAPI, {
            method: "GET",
            url: "/tasks/{id}",
            path: {
                id: id,
            },
            errors: {
                400: `Invalid request, including the errors`,
                404: `Task not found, including the errors`,
            },
        });
    }
    /**
     * Update an existing task
     * Update certain properties of an existing task. This will save the task in Dart for later access, search, etc. Any properties that are not specified will not be changed.
     * @param id
     * @param requestBody
     * @returns WrappedTask Success, including the updated task
     * @throws ApiError
     */
    static updateTask(id, requestBody) {
        return (0, request_1.request)(OpenAPI_1.OpenAPI, {
            method: "PUT",
            url: "/tasks/{id}",
            path: {
                id: id,
            },
            body: requestBody,
            mediaType: "application/json",
            errors: {
                400: `Invalid request, including the errors`,
                404: `Task not found, including the errors`,
            },
        });
    }
    /**
     * Delete an existing task
     * Move an existing task to the trash, where it can be recovered if needed. Nothing else about the task will be changed.
     * @param id
     * @returns WrappedTask Success, including the deleted task
     * @throws ApiError
     */
    static deleteTask(id) {
        return (0, request_1.request)(OpenAPI_1.OpenAPI, {
            method: "DELETE",
            url: "/tasks/{id}",
            path: {
                id: id,
            },
            errors: {
                400: `Invalid request, including the errors`,
                404: `Task not found, including the errors`,
            },
        });
    }
    /**
     * List all tasks that the user has access to. This will return a list of tasks, including the title, dartboard, status, description and others.
     * @returns PaginatedConciseTaskList
     * @throws ApiError
     */
    static listTasks({ assignee, assigneeId, dartboard, dartboardId, description, dueAtAfter, dueAtBefore, ids, inTrash, isCompleted, limit, offset, priority, size, startAtAfter, startAtBefore, status, statusId, tag, tagId, title, type, typeId, }) {
        return (0, request_1.request)(OpenAPI_1.OpenAPI, {
            method: "GET",
            url: "/tasks/list",
            query: {
                assignee: assignee,
                assignee_id: assigneeId,
                dartboard: dartboard,
                dartboard_id: dartboardId,
                description: description,
                due_at_after: dueAtAfter,
                due_at_before: dueAtBefore,
                ids: ids,
                in_trash: inTrash,
                is_completed: isCompleted,
                limit: limit,
                offset: offset,
                priority: priority,
                size: size,
                start_at_after: startAtAfter,
                start_at_before: startAtBefore,
                status: status,
                status_id: statusId,
                tag: tag,
                tag_id: tagId,
                title: title,
                type: type,
                type_id: typeId,
            },
        });
    }
}
exports.TaskService = TaskService;
