// dart-tools/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ViewService = exports.TaskService = exports.FolderService = exports.DocService = exports.DartboardService = exports.ConfigService = exports.CommentService = exports.Priority = exports.OpenAPI = exports.CancelError = exports.CancelablePromise = exports.ApiError = void 0;
/* generated using openapi-typescript-codegen -- do not edit */
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
var ApiError_1 = require("./core/ApiError");
Object.defineProperty(exports, "ApiError", { enumerable: true, get: function () { return ApiError_1.ApiError; } });
var CancelablePromise_1 = require("./core/CancelablePromise");
Object.defineProperty(exports, "CancelablePromise", { enumerable: true, get: function () { return CancelablePromise_1.CancelablePromise; } });
Object.defineProperty(exports, "CancelError", { enumerable: true, get: function () { return CancelablePromise_1.CancelError; } });
var OpenAPI_1 = require("./core/OpenAPI");
Object.defineProperty(exports, "OpenAPI", { enumerable: true, get: function () { return OpenAPI_1.OpenAPI; } });
var Priority_1 = require("./models/Priority");
Object.defineProperty(exports, "Priority", { enumerable: true, get: function () { return Priority_1.Priority; } });
var CommentService_1 = require("./services/CommentService");
Object.defineProperty(exports, "CommentService", { enumerable: true, get: function () { return CommentService_1.CommentService; } });
var ConfigService_1 = require("./services/ConfigService");
Object.defineProperty(exports, "ConfigService", { enumerable: true, get: function () { return ConfigService_1.ConfigService; } });
var DartboardService_1 = require("./services/DartboardService");
Object.defineProperty(exports, "DartboardService", { enumerable: true, get: function () { return DartboardService_1.DartboardService; } });
var DocService_1 = require("./services/DocService");
Object.defineProperty(exports, "DocService", { enumerable: true, get: function () { return DocService_1.DocService; } });
var FolderService_1 = require("./services/FolderService");
Object.defineProperty(exports, "FolderService", { enumerable: true, get: function () { return FolderService_1.FolderService; } });
var TaskService_1 = require("./services/TaskService");
Object.defineProperty(exports, "TaskService", { enumerable: true, get: function () { return TaskService_1.TaskService; } });
var ViewService_1 = require("./services/ViewService");
Object.defineProperty(exports, "ViewService", { enumerable: true, get: function () { return ViewService_1.ViewService; } });
